src/peft/utils/save_and_load.py

```diff
@@ -197,7 +197,14 @@ def renamed_dora_weights(k):
     # ADDITIONAL TRAINING MODULES / MODULES_TO_SAVE
     for name, module in model.named_modules():
         if isinstance(module, AuxiliaryTrainingWrapper):
-            to_return.update({f"{name}.{k}": v for k, v in module.adapter_state_dict(adapter_name).items()})
+            # Compute the module-relative state dict to make it easier for the adapter to fetch the appropriate
+            # keys that the module thinks need to be saved. We cannot rely on `.state_dict()` internally of the
+            # module since accelerators like DeepSpeed require special handling which is done for the model
+            # state dict from above but most likely not in the module itself. See #2450.
+            module_state_dict = {k[len(f"{name}.") :]: v for k, v in state_dict.items() if k.startswith(f"{name}.")}
```

Comment:
Since we're Python 3.9+, would [`k.removeprefix`](https://docs.python.org/3/library/stdtypes.html#str.removeprefix)  work?